ai/langchain4j/src/test/java/io/quarkus/ts/langchain4j/AbstractChatbotIT.java

```diff
@@ -0,0 +1,114 @@
+package io.quarkus.ts.langchain4j;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.WebSocket;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import org.awaitility.Awaitility;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.Protocol;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.annotations.DisabledOnNative;
+import io.quarkus.test.services.URILike;
+
+@DisabledOnNative(reason = "https://issues.redhat.com/browse/QUARKUS-6774")
+public abstract class AbstractChatbotIT {
+    protected static final Logger LOG = Logger.getLogger(ChatbotIT.class);
+    static final String DEFAULT_ARGS = "--no-transfer-progress -DskipTests=true -DskipITs=true -Dquarkus.platform.version=${QUARKUS_PLATFORM_VERSION} -Dquarkus.platform.group-id=${QUARKUS_PLATFORM_GROUP-ID}";
+    private List<String> answers;
+    private WebSocket webSocket;
+
+    static String getKey() {
+        return ConfigProvider.getConfig().getValue("quarkus.langchain4j.openai.api-key", String.class);
+    }
+
+    @BeforeEach
+    void setUp() throws ExecutionException, InterruptedException {
+        String socket = "/chatbot";
+        URILike uri = getApp().getURI(Protocol.WS).withPath(socket);
+        LOG.info("Connecting to: " + uri);
+        answers = Collections.synchronizedList(new ArrayList<>());
+        webSocket = HttpClient.newHttpClient().newWebSocketBuilder()
+                .buildAsync(URI.create(uri.toString()), new ChatbotIT.WebSocketListener(answers)).get();
+    }
+
+    abstract RestService getApp();
+
+    @AfterEach
+    void tearDown() throws ExecutionException, InterruptedException {
+        answers = null;
+        webSocket.sendClose(WebSocket.NORMAL_CLOSURE, "").get();
+    }
+
+    @Test
+    public void smoke() {
```

Comment:
Shouldn't this test be always executed as first? I don't know if test ordering matters too much here, but seems to me, that this should be ordered, for this test to always execute first.

Reply:
Ordering should not matter, since we start new dialogue each time. We just ignore this message in other tests ( "answers.size() <= 1").

Regarding 30 seconds timeout: given, that we connect to third-party API here, it is not impossible. On the other hand, since we have total test timeout for the 10 seconds in the other test, I suppose it doesn't make sense to wait longer here. Changed the value.